fulp_modules/QM_wrestling/code/wrestling.dm

```diff
@@ -0,0 +1,455 @@
+			user.adjustBruteLoss(step)
+			return
+
+		target.visible_message("<span class='warning'>[user] swings [target] directly into [collateral_mob], sending [collateral_mob.p_them()] flying!</span>", \
+			"<span class='userdanger'>You're smacked into [collateral_mob]!</span>", ignored_mobs = collateral_mob)
+		to_chat(collateral_mob, "<span class='userdanger'>[user] swings [target] directly into you, sending you flying!</span>")
+
+		collateral_mob.adjustBruteLoss(step*0.5)
+		collateral_mob.throw_at(collat_throw_target, round(step * 0.25) + 1, round(step * 0.25) + 1)
+		step -= 5
+		delay += 5
+
+	step++
+	if(step >= WRESTLING_THROW_STEPS)
+		finish_swing(user, target, original_dir)
+	else if(step < 0)
+		user.visible_message("<span class='danger'>[user] loses [user.p_their()] momentum on [target]!</span>", "<span class='warning'>You lose your momentum on swinging [target]!</span>", ignored_mobs = target)
+		to_chat(target, "<span class='userdanger'>[user] loses [user.p_their()] momentum and lets go of you!</span>")
+	else
+		addtimer(CALLBACK(src, .proc/swing_loop, user, target, step, original_dir), delay)
+
+/// Time to toss the victim at high speed
+/datum/martial_art/fulpwrestling/proc/finish_swing(mob/living/user, mob/living/target, original_dir)
+	if(!target || !user || user.incapacitated())
+		return
+	if(get_dist(user, target) > 1 || !isturf(user.loc) || !isturf(target.loc))
+		to_chat(user, "<span class='warning'>You lose your grasp on [target]!</span>")
+		return
+
+	user.setDir(original_dir)
+	target.forceMove(user.loc)
+	target.visible_message("<span class='danger'>[user] throws [target]!</span>", \
+					"<span class='userdanger'>You're thrown by [user]!</span>", "<span class='hear'>You hear aggressive shuffling and a loud thud!</span>", null, user)
+	to_chat(user, "<span class='danger'>You throw [target]!</span>")
+	playsound(user.loc, "swing_hit", 50, TRUE)
+	var/turf/T = get_edge_target_turf(user, user.dir)
+	if(!isturf(T))
+		return
+	if(!target.stat)
+		target.emote("scream")
+	target.throw_at(T, 10, 6, user, TRUE, TRUE)
+	log_combat(user, target, "has thrown (Wrestling)")
+
+#undef WRESTLING_THROW_STEPS
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/// Intents
+/datum/martial_art/fulpwrestling/harm_act(mob/living/user, mob/living/target)
+	if(!can_use(user))
+		return FALSE
+	if(check_streak(user,target))
+		return TRUE
+	log_combat(user, target, "harmed (Wrestling)")
+	..()
+
+/datum/martial_art/fulpwrestling/disarm_act(mob/living/user, mob/living/target)
+	if(!can_use(user))
+		return FALSE
+	if(check_streak(user,target))
+		return TRUE
+	log_combat(user, target, "disarmed (Wrestling)")
+	..()
+
+/datum/martial_art/fulpwrestling/grab_act(mob/living/user, mob/living/target)
+	if(!can_use(user))
+		return FALSE
+	if(check_streak(user,target))
+		return TRUE
+	if(user.pulling == target)
+		return TRUE
```

Comment:
```suggestion
	if(user.pulling == target)
		fulpslam(user, target)
		return TRUE
```
an alternative to my change above; this would let you remove the fulpslam streak, the fulpslam action button, and the pulling check in fulpslam()

Reply:
this is, of course, assuming that wrestlers being unable to aggro grab without taking off their belt is a feature (which is a bit sad, because tableslamming people sounds kind of wrestling-y).